### ISSUE TITLE
F# binding does not have debug symbols

### ISSUE THREAD

**Issue by mhutch** [mono/monodevelop#5672]
The F# binding does not have debug symbols, so stack traces are not very useful

**Comment by nosami** [comment]
This is fixed in latest builds of Mono 5.12, 5.14 and 5.16

**Comment by mhutch** [comment]
There's a little more to it. It appears that although the FSharpBinding.dll itself had mdb symbols, the issue was that the mdb symbols were not being copied into the tests folder.

It looks like Mono's MSBuild no longer copies mdb symbols from local copied project references. /cc @radical

**Comment by nosami** [comment]
Yeah - I noticed that. The only reason that FSharpBinding was still using mdb symbols was because of the same crash that you encountered. Later when I noticed that mdb files weren't being copied to output folders it gave me another incentive to try and fix portable pdb debugging.

All the templates use pdb files by default. Now that I've fixed the issue with the debugger crashing with pdb files  (it only affected assemblies that had pinvokes), I can't think of a reason to support mdb debugging.

**Comment by mhutch** [comment]
@nosami folks might be referencing assemblies with old symbols? Seems bad to break back-compat.

**Comment by nosami** [comment]
It's not that straightforward to build with .mdb symbols. `<DebugType>Full</DebugType>` and `<DebugType>Portable</DebugType>` both produce portable symbols (just like C# on Mono). Anyone who formerly used .mdb symbols was automatically switched over to .pdb some time back. 

You need to use the following to remain on old symbols, that I added as a failsafe and only told to the couple of people that were having issues with debugging. FSharpBinding was unfortunately one of the projects still using mdb files, but there is no longer any reason for this.

```
    <FscDebugType>full</FscDebugType>
    <FscDebugFileExt>.mdb</FscDebugFileExt>
```

The F# compiler produces much better step information / ranges using ppdb than mdb. MDB files should probably still work in theory, I'm just not sure who this would benefit :)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.73 (master)

Pull Request #5673 merged by: Mikayla Hutchinson
Author: mono
Commit: 0bf1f73c8b676c2c415117549ddc954b40f2808a (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.126 (release-7.7)

Pull Request #5673 merged by: Mikayla Hutchinson
Author: mono
Commit: 0bf1f73c8b676c2c415117549ddc954b40f2808a (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.931 (release-7.7)

Pull Request #5673 merged by: Mikayla Hutchinson
Author: mono
Commit: 0bf1f73c8b676c2c415117549ddc954b40f2808a (mono/monodevelop)
